Chat final

nwjbrandon opened the pull request.

nwjbrandon committed:
add chat

nwjbrandon committed:
create barebone of chat list

nwjbrandon committed:
complete input

nwjbrandon committed:
complete chatoffer

nwjbrandon committed:
add marginless

nwjbrandon committed:
styling of chat

nwjbrandon committed:
fix issues with code

nwjbrandon committed:
style with bulma

karrui commented on src/routes/chat/Chat.container.js:
```diff
@@ -0,0 +1,19 @@
+import { fetchChatListAction } from './Chat.ducks';
+import Chat from './Chat';
+
+const matchState = state => {
```
Not going to use them anymore if we stop using connect, but as said before in the previous PR,........ `mapStateToProps` is the convention.

karrui commented on src/routes/chat/SocketService/socketSetup.js:
```diff
@@ -0,0 +1,15 @@
+  Socket.socket = io.connect(`${process.env.REACT_APP_BACKEND_API}chat`);
+};
+
+connect();
```
don't call it here directly, this means once the app starts up the socket will connect. Only call it onEffect in Chat.jsx or something.

karrui replied:
Not sure if this is good, since the socket will automatically connect once this file is imported. Maybe force users to call it manually?

karrui commented on src/routes/chat/Chat.container.js:
```diff
@@ -0,0 +1,19 @@
+import { connect } from 'react-redux';
```
Avoid connect, use `import { useSelector, useDispatch } from 'react-redux'` in `Chat.jsx` itself instead.

karrui commented on src/routes/chat/ChatInput/ChatInput.container.js:
```diff
@@ -0,0 +1,19 @@
+import { connect } from 'react-redux';
```
Don't need connect.

karrui commented on src/routes/chat/ChatInput/ChatInput.jsx:
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import SendIcon from '../../../assets/images/send_icon.png';
```
use absolute paths. `import SendIcon from 'assets/.....';`.
In fact, why do we need this image? Use fontawesome or something: https://fontawesome.com/icons?d=gallery&q=send

Always prefer SVGs over images.

karrui commented on src/routes/chat/Chat.jsx:
```diff
@@ -0,0 +1,56 @@
+import './Chat.scss';
+
+SocketResponseService.initialize();
+Moment.startPooledTimer();
```
!!! don't put it outside of the function. start it on onEffect. Then use onEffect's return to shut it down when the component unmounts. Same for Moment.

karrui commented on src/routes/chat/ChatRoom/ChatRoom.container.js:
```diff
@@ -0,0 +1,16 @@
+import { connect } from 'react-redux';
```
don't need conenct

karrui commented on src/routes/chat/Chat.jsx:
```diff
@@ -0,0 +1,56 @@
+import ChatList from './ChatList/ChatList.container';
+import ChatRoom from './ChatRoom/ChatRoom.container';
+import ChatInput from './ChatInput/ChatInput.container';
+import ChatOffer from './ChatOffer/ChatOffer';
```
All these should be called using their respective `index.js`s and thus don't have the `.container` ending. Also delete the `ChatOffer.container.js` file if it is empty.

karrui reviewed (changes requested):
brief glance, please fix

nwjbrandon committed:
fix issues

nwjbrandon committed:
address consle errors

nwjbrandon committed:
fix bug in the order in chat list

nwjbrandon committed:
camel case formatter

nwjbrandon committed:
fix some comments

nwjbrandon committed:
fix comments

karrui commented on src/routes/chat/ChatList/ChatList.jsx:
```diff
@@ -0,0 +1,17 @@
+  return (
+    <div>
+      {chatList.map(chat => (
+        <ChatItem chat={chat} />
```
Add key

karrui commented on src/routes/chat/ChatOffer/ChatOffer.jsx:
```diff
@@ -0,0 +1,40 @@
+        </h1>
+        <div className="columns is-gapless is-mobile">
+          <div className="column">
+            <button
```
Add issue to this repo and add TODO so we don't forget.
`//TODO(#issueNumber): add make offer button in chat functional`

karrui commented on src/routes/chat/ChatList/ChatItem.jsx:
```diff
@@ -0,0 +1,71 @@
+  return (
+    <div
+      role="presentation"
+      key={chat.chatRoomId}
```
Don't put key here, put the `key` where `.map` is called.

karrui commented on src/routes/chat/ChatRoom/ChatMessage.jsx:
```diff
@@ -0,0 +1,37 @@
+
+const ChatMessage = ({ chat }) => {
+  return (
+    <div key={chat.createdAt} className="columns is-marginless is-mobile">
```
Same, don't put `key` here, put where `.map` is called.

karrui commented on src/routes/chat/ChatRoom/ChatRoom.jsx:
```diff
@@ -0,0 +1,19 @@
+  return (
+    <div>
+      {chatRoom.map(chat => (
+        <ChatMessage chat={chat} />
```
put `key` here

karrui reviewed (approved):
LGTM! Just fix those nits and we can submit >:). Thanks for the patience!

karrui commented:
Fix merge conflicts and my last few comments and we can merge it in :)

When you merge, squash everything (since your commit messages are hideous lol), and follow this git commit guidelines I just added to CONTRIBUTING.md.

```
feat(Chat): add basic chat functionality

Chat works with socket.io to create and maintain websockets.
NOTE:
- Make offers or cancel match buttons in chat room are not functional yet
- There are no new message notifications
```

nwjbrandon committed:
feat(Chat): add basic chat functionality

chat works with socket.io to create and maintain websockets

NOTE:
- Send button is not functional
- No new message notifications
- Offer and cancel buttons is not functional
- No online status

nwjbrandon committed:
Merge branch 'master' into chat-final
